src/workerd/jsg/jsg.h

```diff
@@ -637,6 +637,9 @@ namespace {
 // all in JavaScript when the optional is null in C++ (as opposed to the field being present but
 // assigned the value `undefined`).
 //
+// Note that if a `validate` function is provided, then it will be called after the struct is
+// unwrapped from v8. This would be an appropriate time to throw an error.
```

Comment:
nit: might be good to include the example from the `README.md` here also but up to you.  At the very least let's document the expected signature and indicate that `validate` is expected to throw its own errors and return void.